_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,483 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgments:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][] for which all is true: 
+- the event's attribute `key` is a [printable character][] key; and
+- the event's method `getModifierState` returns `false` for any of the [valid modifier keys][].
+
+## Expectation
+
+For each test target at least one of the following is true:
+ - (**Remap**:) there is at least one [set of clearly labeled instruments][] to [block events][blocked event] that use the [same key][same key events] as the test target and whose `getModifierState` method returns `false` for any of the [valid modifier keys][]; or
+ - (**Focus**:) if the [event target][] does not have a [semantic role][] that inherits from the [abstract role](https://www.w3.org/TR/wai-aria/#abstract_roles) of `widget` the event does not cause [changes in content][] of the [HTML document][].
+
+## Assumptions
+
+- The event listeners listening to [keyboard events][keyboard event] trigger a functionality in the web page. If they do not trigger any such functionality failing this rule might not be a failure of the success criterion.
+- If there are ways to disable the result of [keyboard events][keyboard event] that do not require the user to interact with the web page (e.g. a setting at the operating system level), failing this rule might not be a failure of the success criterion.
+
+## Accessibility Support
+
+Currently [keyboard events][keyboard event] only support the types `keydown` and `keyup`. [Keyboard events][keyboard event] of type `keypressed` are considered [legacy keyboard events][] and are thus ignored by this rule.
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G217 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+- [F99 Failure of Success Criterion 2.1.4 due to implementing character key shortcuts that cannot be turned off or remapped](https://www.w3.org/WAI/WCAG21/Techniques/failures/F99)
+
+## Test Cases
+
+### Passed
```

Comment:
We could have an example where 2 instruments are needed to remap/disable the event. That would illustrate the "set of instruments" (for a single event) part of the expectation.
Something like 2 checkboxes.
I feel this is very contrived and not natural, but that these kinds of cases could occur (e.g., settings are locked by default and require a toggle to unlock, and then the actual setting can be changed).